tests/test_monitor.py

```diff
@@ -34,3 +35,47 @@ def test_monitor():
     assert set(last_logline.keys()) == {'l', 't', 'r'}, "Incorrect keys in monitor logline"
     file_handler.close()
     os.remove(mon_file)
+
+def test_monitor_load_results(tmp_path):
+    """
+    test load_results on log files produced by the monitor wrapper
+    """
+    env1 = gym.make("CartPole-v1")
+    env1.seed(0)
+    monitor_file1 = os.path.join(tmp_path, "stable_baselines-test-{}.monitor.csv".format(uuid.uuid4()))
+    monitor_env1 = Monitor(env1, monitor_file1)
+
+    monitor_files = get_monitor_files(tmp_path)
+    assert len(monitor_files) == 1
+    assert monitor_file1 in monitor_files
+
+    monitor_env1.reset()
+    for _ in range(1000):
+        _, _, done, _ = monitor_env1.step(monitor_env1.action_space.sample())
+        if done:
+            monitor_env1.reset()
+
+    results_size1 = load_results(os.path.join(tmp_path)).size
+    assert results_size1 > 0
+
+    env2 = gym.make("CartPole-v1")
+    env2.seed(0)
+    monitor_file2 = os.path.join(tmp_path, "stable_baselines-test-{}.monitor.csv".format(uuid.uuid4()))
+    monitor_env2 = Monitor(env2, monitor_file2)
+    monitor_files = get_monitor_files(tmp_path)
+    assert len(monitor_files) == 2
+    assert monitor_file1 in monitor_files
+    assert monitor_file2 in monitor_files
+
+    monitor_env2.reset()
+    for _ in range(1000):
+        _, _, done, _ = monitor_env2.step(monitor_env2.action_space.sample())
+        if done:
+            monitor_env2.reset()
+
+    results_size2 = load_results(os.path.join(tmp_path)).size
+
+    assert results_size2 > results_size1
```

Comment:
why is that?

Reply:
The number of entries in the log file is less predictable when sampling from the action space, I guess.

It is not longer twice the size. results_size1 is 168 and results_size2 is 332 when I run and the numbers change between runs. It's fragile to depend too much random behavior in a test.

Reply:
Counting episodes now.